### PR TITLE
config hypervisor

### LINKED ISSUE #826: custom machine accelerators and firmwares
Allow users specify custom machine accelerators and firmwares

### PR DESCRIPTION
**Pull request by devimc**
Add options to specify custom firmwares and machine accelerators

fixes #826

### CODE REVIEW

File: config/configuration.toml.in [hunk 1]
```diff
@@ -10,6 +10,13 @@ machine_type = "@MACHINETYPE@"
 # trouble running pre-2.15 glibc
 kernel_params = "@KERNELPARAMS@"
 
+# Path to the firmware.
+# If you want that qemu uses the default firmware leave this option empty
+firmware = "@FIRMWAREPATH@"
+
+# Machine accelerators
+machine_accelerators="@MACHINEACCELERATORS@"
```
Comment (jodh-intel) [inline config/configuration.toml.in]: What format is this variable? Can you add a comment explaining if the accelerators should be space/comma/whatever-separated, ideally with an example showing >1 accelerator being specified.

Comment (devimc) [reply config/configuration.toml.in]: sure, thanks :)

File: config/configuration.toml.in [hunk 2]
```diff
@@ -10,6 +10,15 @@ machine_type = "@MACHINETYPE@"
...
 kernel_params = "@KERNELPARAMS@"
 
+# Path to the firmware.
+# If you want that qemu uses the default firmware leave this option empty
+firmware = "@FIRMWAREPATH@"
+
+# Machine accelerators
+# space-separated list of machine accelerators to pass to the hypervisor.
+# For example, `machine_accelerators = "nosmm,nosmbus,nosata,nopit,static-prt,nofw"`
+machine_accelerators="@MACHINEACCELERATORS@"
```
Comment (jodh-intel) [inline config/configuration.toml.in]: Thanks - but that should be "comma-separated" right?

----
Aside: it's worth pointing out that this is the first config option that takes a comma-separated list. I think the current approach (where minimal validation is performed on the list) is fine, but if a user sets the variable to something like:

```
machine_accelerators = "foo,,"
```

... I suspect qemu will get pretty unhappy (but I haven't tested it).

Comment (devimc) [reply config/configuration.toml.in]: ohh yep, sorry :(

Comment (devimc) [reply config/configuration.toml.in]: Fixed, thanks @jodh-intel

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**